service-registry/registry-zero-config/src/main/java/org/apache/servicecomb/zeroconfig/ZeroConfigRegistryApplicationListener.java

```diff
@@ -52,24 +52,23 @@
   private static final Logger LOGGER = LoggerFactory
       .getLogger(ZeroConfigRegistryApplicationListener.class);
 
-  private ApplicationContext applicationContext;
   private MulticastSocket multicastSocket;
 
   @Override
   public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
 
     // same mechanism as Local registry to enable the Zero Config registry
-    String flag = System.getProperty(ZeroConfigRegistryConstants.ZERO_CONFIG_REGISTRY_FLAG);
-    if (flag != null && flag.equals(ZeroConfigRegistryConstants.ZERO_CONFIG_REGISTRY_ENABLE_FLAG)) {
-      this.applicationContext = applicationContext;
-      BeanUtils.setContext(applicationContext);
+    boolean enable = DynamicPropertyFactory.getInstance()
+        .getBooleanProperty("servicecomb.zeroconfig.enabled", false).get();
+
+    if (enable) {
       ServerUtil.init();
       ClientUtil.init();
 
       try {
         this.multicastSocket = new MulticastSocket();
       } catch (IOException e) {
-        LOGGER.error("Failed create MulticastSocket object", e);
+        LOGGER.error("Failed to create MulticastSocket object", e);
```

Comment:
better to throw exception and prevent bootup

Reply:
agreed.